src/azure-cli/azure/cli/command_modules/policyinsights/custom.py

```diff
@@ -565,3 +731,21 @@ def _remove_leading_and_trailing_slash(s):
             s = s[:-1]
 
     return s
+
+
+def _get_policy_assignment_id(cmd, policy_assignment):
+    # Get the full resource ID of the referenced policy assignment
+    if (not is_valid_resource_id(policy_assignment) and
+            not policy_assignment.lower().startswith("/providers/microsoft.management/managementgroups/")):
+        from ._client_factory import cf_policy
+        policy_assignment_client = cf_policy(cmd.cli_ctx).policy_assignments
+        policy_assignments = policy_assignment_client.list()
+        policy_assignment_ids = [
+            p.id for p in policy_assignments if p.name.lower() == policy_assignment.lower()]
+        if not policy_assignment_ids:
+            raise CLIError(
```

Comment:
Please use the specific error type `ResourceNotFoundError` instead of `CLIError`